02-express-tutorial/app.js

```diff
@@ -1 +1,58 @@
-console.log('Express Tutorial')
+const express = require('express');
+const app = express();
+const { products } = require('./data');
+
+app.use(express.static('./public'));
+
+app.get('/api/v1/test', (req, res) => {
+  res.json({ message: 'It worked!' });
+});
+
+app.get('/api/v1/products', (req, res) => {
+  res.json(products);
+});
+
+app.get('/api/v1/products/:productID', (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find(p => p.id === idToFind);
+
+  if (!product) {
+    return res.status(404).json({ message: 'That product was not found' });
+  }
+
+  return res.json(product);
+});
+
+app.get('/api/v1/query', (req, res) => {
+  const { search, limit, priceLimit } = req.query;
+  let foundedProducts = [...products];
```

Comment:
Nice use of the spread operator!

Reply:
Yes, you are right. I'll fix it now. Thank you.